client/src/components/secretMessage.test.tsx

```diff
@@ -0,0 +1,82 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import React from 'react';
+import {
+  render, screen,
+} from '@testing-library/react';
+import axios from 'axios';
+import SecretMessage from './SecretMessage';
+
+test('render the secret body when there is an valid id', async () => {
+  // Arrange
+  const fakeId = '';
+  const res = {
+    status: 200,
+    data: {
+      id: fakeId,
+      body: 'test',
+    },
+  };
+  // Act
+  jest.spyOn(axios, 'get').mockImplementation(() => Promise.resolve(res));
+  render(<SecretMessage id={fakeId} />);
+  // Assert
+  const element = await screen.findByText(/test/);
+  expect(element).toBeInTheDocument();
+});
+
+test('render an error message when there is an invalid id', async () => {
+  // Arrange
+  const fakeId = '';
+  const error = {
+    response: {
+      status: 400,
+      data: {
+        message: 'Bad Request',
+      },
+    },
+  };
+  // Act
+  jest.spyOn(axios, 'get').mockImplementation(() => Promise.reject(error));
+  render(<SecretMessage id={fakeId} />);
+  // Assert
+  const element = await screen.findByText(/Secret id is not valid/);
+  expect(element).toBeInTheDocument();
+});
+
+test('render an error message when id not exist or has been viewed', async () => {
+  // Arrange
+  const fakeId = '';
+  const error = {
+    response: {
+      status: 404,
+      data: {
+        message: 'Not Found',
+      },
+    },
+  };
+  // Act
+  jest.spyOn(axios, 'get').mockImplementation(() => Promise.reject(error));
+  render(<SecretMessage id={fakeId} />);
+  // Assert
+  const element = await screen.findByText(/Secret doesn't exist or has been viewed/);
```

Comment:
The different server response should be handled differently in the log (which we don't worry about it now). But not in UI. Feel free water your humour here. The most important thing is to tell the user what to do next.

```suggestion
  const element = await screen.findByText(/Can you keep a secret? So do I./);
```

Reply:
This one is about the wording. Still, it's a suggestion, not a recommendation.